collective/pfg/signup/adapter.py

```diff
@@ -214,13 +214,10 @@ def __init__(self, oid, **kwargs):
         """Initialize class."""
         FormActionAdapter.__init__(self, oid, **kwargs)
         self.waiting_list = OOBTree()
-        self.mail_settings = IMailSchema(getUtility(ISiteRoot))
         
-    # @property
-    # def mail_settings(self):
-    #     # registry = getUtility(IRegistry)
-    #     #return registry.forInterface(IMailSchema, prefix='plone')
-    #     return IMailSchema(self.context)
+    @property
+    def mail_settings(self):
+        return IMailSchema(getUtility(ISiteRoot))
```

Comment:
@djay What was preventing it from getting the mail schema within `__init__` that means this solves it?

Reply:
because it was a change that happened after teh adapter was created then the __init__ is never called. It's an upgrade step problem